hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestAbfsPaginatedDelete.java

```diff
@@ -0,0 +1,279 @@
+    AbfsRestOperationException e = intercept(AbfsRestOperationException.class, () ->
+        client.getPathStatus(testPath.toString(), false, testTracingContext, null));
+    Assertions.assertThat(e.getStatusCode())
+        .describedAs("Path should have been deleted").isEqualTo(HTTP_NOT_FOUND);
+  }
+
+  public void testNonRecursiveDeleteWithPaginationInternal(boolean isPaginatedDeleteEnabled) throws Exception{
+    final AzureBlobFileSystem fs = isHnsEnabled ? this.firstTestUserFs : getFileSystem();
+    TracingContext testTracingContext = getTestTracingContext(this.firstTestUserFs, true);
+    Path testPath = new Path("/emptyPath");
+    fs.mkdirs(testPath);
+
+    // Set the paginated enabled value and xMsVersion at client level.
+    AbfsClient client = fs.getAbfsStore().getClient();
+    client.getAbfsConfiguration().setIsPaginatedDeleteEnabled(isPaginatedDeleteEnabled);
+    AbfsRestOperation op = client.deletePath(testPath.toString(), false, null, testTracingContext);
+
+    // Getting the url that was used to make the request
+    String urlUsed = op.getUrl().toString();
+
+    // Assert that paginated query param was not set to make request
+    Assertions.assertThat(urlUsed)
+          .describedAs("Url must not have paginated as query param")
+          .doesNotContain(QUERY_PARAM_PAGINATED);
+
+    // Assert that deletion was successful in every scenario.
+    AbfsRestOperationException e = intercept(AbfsRestOperationException.class, () ->
+        client.getPathStatus(testPath.toString(), false, testTracingContext, null));
+    Assertions.assertThat(e.getStatusCode())
+        .describedAs("Path should have been deleted").isEqualTo(HTTP_NOT_FOUND);
+  }
+
+  public void testRecursiveDeleteWithInvalidCTInternal(boolean isPaginatedEnabled) throws Exception {
+    final AzureBlobFileSystem fs = isHnsEnabled ? this.firstTestUserFs : getFileSystem();
+    Path smallDirPath = createSmallDir();
+    String randomCT = "randomContinuationToken1234";
+    TracingContext testTracingContext = getTestTracingContext(this.firstTestUserFs, true);
+
+    AbfsClient client = fs.getAbfsStore().getClient();
+    client.getAbfsConfiguration().setIsPaginatedDeleteEnabled(isPaginatedEnabled);
+
+    AbfsRestOperationException e = intercept(AbfsRestOperationException.class, () ->
+        client.deletePath(smallDirPath.toString(), true, randomCT, testTracingContext));
+    Assertions.assertThat(e.getStatusCode())
+        .describedAs("Request Should fail with 400").isEqualTo(HTTP_BAD_REQUEST);
+  }
+
+  private void setFirstTestUserFsAuth() throws IOException {
+    if (this.firstTestUserFs != null) {
+      return;
+    }
+    checkIfConfigIsSet(FS_AZURE_ACCOUNT_OAUTH_CLIENT_ENDPOINT
+        + "." + getAccountName());
+    Configuration conf = getRawConfiguration();
+    setTestFsConf(FS_AZURE_BLOB_FS_CLIENT_ID, FS_AZURE_BLOB_FS_CHECKACCESS_TEST_CLIENT_ID);
+    setTestFsConf(FS_AZURE_BLOB_FS_CLIENT_SECRET,
```

Comment:
let add clientId and secret in the method args. Reason being, we always want `setFirstTestUserFsAuth` and `setDefaultAclOnRoot` in sync.

Reply:
Are UID, clientId, and secret not related.  if yes, requesting you to supply clientId and secret from the `setup()` method.